app/backend/approaches/tabulardataassistant.py

```diff
@@ -59,126 +59,74 @@
 def refreshagent():
     global pdagent
     pdagent = None
-
-
-
+def get_image_data(image_path):
+    with Image.open(image_path) as img:
+        img_byte_arr = io.BytesIO()
+        img.save(img_byte_arr, format='PNG')
+        img_byte_arr = img_byte_arr.getvalue()
+        img_base64 = base64.b64encode(img_byte_arr)
+    return img_base64.decode('utf-8')
+
+def save_chart(query):
+    temp_dir = tempfile.gettempdir()
+    q_s = f' If any charts or graphs or plots were created save them in the {temp_dir} directory".'
+    query += ' . '+ q_s
+    return query
+
+def get_images_in_temp():
+    temp_dir = tempfile.gettempdir()
+    image_files = glob.glob(os.path.join(temp_dir, '*.[pjJ][npNP][gG]*'))
+    image_data = [get_image_data(file) for file in image_files]
+
+    # Delete the files after reading them
+    for file in image_files:
+        os.remove(file)
+        
+    return image_data
  
 def save_df(dff):
     global dffinal
     dffinal = dff      
 
 
-def is_base64(s):
-    pattern = r'^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$'
-    if s.startswith('base64 string:') and s.endswith('end of base 64 string'):
-        return True
-    elif bool(re.fullmatch(pattern, s)):
-        return True
-    return False
-
 
       
 # function to stream agent response 
 def process_agent_scratch_pad(question, df):
     chat = AzureChatOpenAI(
                 openai_api_version=OPENAI_API_VERSION,
                 deployment_name=OPENAI_DEPLOYMENT_NAME)
-    
-    pdagent = create_pandas_dataframe_agent(chat, df, verbose=True,handle_parsing_errors=True,agent_type=AgentType.OPENAI_FUNCTIONS)
-    messages = []
+    if 'chart' or 'charts' or 'graph' or 'graphs' or 'plot' or 'plt' in question:
+        question = save_chart(question)
+    pdagent = create_pandas_dataframe_agent(chat, df, verbose=True,handle_parsing_errors=True,agent_type=AgentType.OPENAI_FUNCTIONS, save_charts=True)
```

Comment:
I can find zero documentation or source for this "save_charts" parameter of langchain. I want to understand what it is doing before we allow it to be used.

Reply:
This appears to save charts to a default directory of /exports/charts unless a save_charts_path is specified.
https://github.com/Sinaptik-AI/pandas-ai/issues/174#issuecomment-1565596501
https://docs.pandas-ai.com/en/stable/examples/

Reply:
YesThere is no reference of Save_Charts in official Langchain docs ,Above link is for PandasAI which is another library and  it works with langahin as well.people have been using this as parameter for pandas dataframe agent and it seem to be functioning same as PandasAI, Saving chart in local directory. If we are blowing away chart after every session then this may pass RAI test.